apps/veil/src/pages/tournament/api/use-index-by-address.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from '@tanstack/react-query';
+import { Address } from '@penumbra-zone/protobuf/penumbra/core/keys/v1/keys_pb';
+import { uint8ArrayToBase64 } from '@penumbra-zone/types/base64';
+import { ViewService } from '@penumbra-zone/protobuf';
+import { connectionStore } from '@/shared/model/connection';
+import { penumbra } from '@/shared/const/penumbra';
+
+/**
+ * Takes and address and checks if it belongs to user
+ */
+export const useIndexByAddress = (address?: Address) => {
```

Comment:
This implementation is ok for now, but we should avoid checking each address in the leaderboard, and instead use the fact that we know the address we've voted with, fetch that, and check if it's among that set. We can then revalidate that query every epoch or something.